### DISCUSSION TITLE
Can't open read-only db file

### DISCUSSION THREAD

**Discussion by glgraca** [pawelsalawa/sqlitestudio#4706] [Q&A]
I would like to query a file which is read-only, but sqlitestudio won't let me.

This database is only for querying old data, so I would like to maintain the file read-only.

Could this be modified or is there some setting I've missed?

Thanks!

**Comment by pawelsalawa** [comment]
What does it tell you when you try to open it?

**Reply by pawelsalawa** [reply]
And which SQLiteStudio version are you using?

**Reply by glgraca** [reply]
I just downloaded it this morning: v3.4.3. It says (I'm translating from Portuguese) "This file is marked as read-only. Try again with another file name".

**Reply by pawelsalawa** [reply]
And what operating system is it?

**Reply by glgraca** [reply]
It's Windows 10. The file is in a network share, but the behaviour is the same if it's a local drive.

I can add the database if it's not readonly and it will let me open it later, even if I've made it readonly again.

So the issue only happens the moment I try to add the database.

**Reply by pawelsalawa** [reply]
This is a surprise. It's been a long time since I ever tested opening read-only db files.

Anyway, the issue is only in the file selector dialog. There is a workaround - you can type the path to your file directly to the file input in Database Dialog (and not select it through file picker) and it will work just fine.

I've created issue #4707 to fix it.

**Reply by glgraca** [reply]
Thank you very much. The workaround will do nicely.

Good job, by the way; the app is very useful.

**Reply by pawelsalawa** [reply]
Note, that using SQLite databases over a network shared drives is not recommended by SQLite itself. It is known to cause unexpected troubles. Just keep that in mind.
